daemon/api.go

```diff
@@ -1481,6 +1487,36 @@ func setSnapConf(c *Command, r *http.Request, user *auth.UserState) Response {
 	return AsyncResponse(nil, &Meta{Change: change.ID()})
 }
 
+func getSnapHistory(c *Command, r *http.Request, user *auth.UserState) Response {
+	vars := muxVars(r)
+	name := vars["name"]
+
+	route := c.d.router.Get(snapCmd.Path)
+	if route == nil {
+		return InternalError("cannot find route for snaps")
+	}
+
+	url, err := route.URL("name", name)
+	if err != nil {
+		return InternalError("cannot build URL for %q snap: %v", name, err)
+	}
+
+	var history []map[string]interface{}
```

Comment:
instead of just declaring `history`, wait until you've gotten `snapRevisions` and use `history := make([]map[string]interface{}, len(snapRevisions)` to pre-allocate it to the right size (and avoid the `append`)